src/projections/adams.cpp

```diff
@@ -33,7 +38,8 @@
 #include "proj_internal.h"
 
 PROJ_HEAD(guyou, "Guyou") "\n\tMisc Sph No inv";
-PROJ_HEAD(peirce_q, "Peirce Quincuncial") "\n\tMisc Sph No inv";
+PROJ_HEAD(peirce_q, "Peirce Quincuncial (Square)") "\n\tMisc Sph No inv";
+PROJ_HEAD(peirce_q_d, "Peirce Quincuncial (Diamond)") "\n\tMisc Sph No inv";
```

Comment:
I'm wondering if we need a new projection method when the difference is only the final rotation. A possiblity would be to have a xy_rot= parameter, like in src/projections/healpix.cpp  .     peirce_q would be obtained with xy_rot=45 and peirce_q_d with xy_rot=0. The question would be what should be the default for perice_q : 45 to have backward compatibility or 0 if it makes more sense as a default ? Changing to 0 could probably be acceptable for PROJ 9, because this projection method is rather new

Reply:
Thank very much for the review + input.  

Can see the argument for using a xy_rot value rather than separate projection if this is better general proj practice, though I'm not exactly sure how to implement that in code.

Alternative fpr code simplicity would be to use a simple option param like ESRI (e.g. 0=square, 1=diamond) rather than angle, to avoid arbitrary angles.

Would strongly vote for the default being the quincunx square (peirce_q / xy_rot=45) or diamond (peirce_q_d / xy_rot=0). This is technically not backward compatible with the proj implementation until this fix (which only produced the inner square of the full diamond, so is xy_rot=0), but a square is surely most likely output wanted.

Reply:
One additional thought on the question of `xy_rot=45/0` vs `type=0/1`. 

I was planning to build on this code to add the lateral quincuncial projection (i.e. hemispheres plotted side-by-side as rectangle). I had originally thought of making it a seperate projection with options for sliding -- to allow all continents to fit.  

But following discussion above, would this be better indicated as param in a single peirce_q proj? e.g. `type=2`?

Reply:
type= would also be fine for me. I'd prefer string values like square, diamond, etc. for clarity. An example of reading a string valued parameter is at https://github.com/OSGeo/PROJ/blob/master/src/transformations/helmert.cpp#L527 . You would want to do that in the setup() function

Reply:
implemented in later commit